oap-server/analyzer/log-analyzer/src/main/java/org/apache/skywalking/oap/log/analyzer/dsl/spec/AbstractSpec.java

```diff
@@ -49,4 +52,13 @@ public void abort(final Closure<Void> cl) {
     public Object propertyMissing(final String name) {
         return BINDING.get().getVariable(name);
     }
+
+    @SuppressWarnings("unused")
+    public String tag(String key) {
+        return BINDING.get().log().getTags().getDataList()
+                .stream()
+                .filter(data -> key.equals(data.getKey()))
+                .map(KeyStringValuePair::getValue)
+                .collect(Collectors.toList()).get(0);
+    }
```

Comment:
You need to make sure this doesn't throw

```suggestion
    @SuppressWarnings("unused")
    public String tag(String key) {
        return BINDING.get().log().getTags().getDataList()
                .stream()
                .filter(data -> key.equals(data.getKey()))
                .map(KeyStringValuePair::getValue)
                .findFirst()
                .orElse("");
    }
```